Lombiq.Tests.UI/Extensions/BasicOrchardFeaturesTestingUITestContextExtensions.cs

```diff
@@ -24,12 +24,21 @@ public static class BasicOrchardFeaturesTestingUITestContextExtensions
     /// <para>The test method assumes that the site is not set up.</para>
     /// </summary>
     /// <param name="setupRecipeId">The ID of the recipe to be used to set up the site.</param>
+    /// <param name="customPageHeaderCheckAsync">
+    /// The custom page header check logic to locate and/or check the header's text. This ultimately gets passed to
+    /// TestContentOperationsAsync().
+    /// </param>
     /// <returns>The same <see cref="UITestContext"/> instance.</returns>
-    public static Task TestBasicOrchardFeaturesAsync(this UITestContext context, string setupRecipeId) =>
-        context.TestBasicOrchardFeaturesAsync(new OrchardCoreSetupParameters(context)
-        {
-            RecipeId = setupRecipeId,
-        });
+    public static Task TestBasicOrchardFeaturesAsync(
+        this UITestContext context,
+        string setupRecipeId,
+        Func<UITestContext, Task> customPageHeaderCheckAsync = null) =>
```

Comment:
I used `Func<UITestContext, Task>` instead of `Action<UITestContext>`. This way the check can be async too and if not, we can put `return Task.CompletedTask;` at the end. I don't think it would work the other way around (using await in a non async function without any warning).

Reply:
Yeah, that's good.